#23 버스 터미널 지역이름으로 조회

0hun opened the pull request:
버스터미널 지역이름으로 조회메소드 추가
객체를 파싱하도록 objectMapper 설정 추가
redis에 저장하는할때는 ResponseEntity가 아닌 해당 객체로 저장을 해야하므로 수정

0hun committed:
#23 버스 터미널 지역이름으로 조회

버스터미널 지역이름으로 조회메소드 추가
객체를 파싱하도록 objectMapper 설정 추가
redis에 저장하는할때는 ResponseEntity가 아닌 해당 객체로 저장을 해야하므로 수정

0hun committed:
#23 버스 터미널 지역이름으로 조회

지역 이름으로 버스터미널 조회 메소드 결과 리턴시 optional로 받도록 수정

f-lab-dev commented on src/main/java/com/younghun/hibusgo/config/RedisConfig.java:
```diff
@@ -16,6 +17,19 @@
+   */
+  @Bean
+  public ObjectMapper objectMapper() {
+    ObjectMapper mapper = new ObjectMapper();
```
스프링에는 기본 ObjectMapper가 bean으로 등록되어있습니다~

0hun replied:
넵 수정하겠습니다

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/BusTerminalController.java:
```diff
@@ -25,27 +24,31 @@
+   * @param region 지역 이름
+   * @return List<BusTerminal>
+   */
+  @Cacheable(value = "terminals.region", key = "#region", cacheManager = "redisCacheManager")
```
일반적으로 controller layer에 캐싱을 달지 않습니다. service에만 캐싱을 넣어주세요~

0hun replied:
넵 수정하겠습니다

f-lab-dev commented on src/main/java/com/younghun/hibusgo/domain/BusTerminal.java:
```diff
@@ -1,13 +1,17 @@
 @Builder
 @ToString
+@AllArgsConstructor
+@NoArgsConstructor
```
빈 객체는 어떤것때문에 만들려고 하시는걸까요?

0hun replied:
ObjectMapper로 역직렬화시 객체 생성 및 값 매핑을 위해 필요해서 사용했습니다.

0hun replied:
https://blog.leocat.kr/notes/2018/09/02/lombok-using-builder-and-noargsconstructor-together

0hun committed:
#23 리뷰 반영

스프링에는 기본 ObjectMapper가 bean으로 등록되어있기 때문에 설정에서 삭제
서비스 레이어에서 캐시하도록 수정 및 만료시간 설정

f-lab-dev commented:
이쪽은 추가 개선을 고민중이신듯 하여 다음 커밋이 올라오면 리뷰하겠습니다~

0hun committed:
#23 캐시 설정

@Scheduled는 파라미터가 없는 메소드에만 적용가능하여, 공통 캐시 설정으로 수정
터미널 정보는 변하지 않는 정보이기 때문에 공통 캐시 설정을 적용

0hun committed:
Merge branch 'issue/22' into issue/23

0hun committed:
#24 리뷰 반영

리스트 값이 없다면 빈 리스트를 넘겨주도록 수정
캐시 및 조회 결과값은 ResponseEntity의 body를 통해 넘겨주도록 수정

0hun committed:
Merge branch 'issue/23' of https://github.com/f-lab-edu/hi-bus-go into issue/23

0hun committed:
#23 터미널 정보 없을시 http 404 return

데이터가 없을 경우 404 리턴 하도록 수정
